cpp/src/Ice/CollocatedRequestHandler.cpp

```diff
@@ -160,7 +160,7 @@ CollocatedRequestHandler::invokeAsyncRequest(OutgoingAsyncBase* outAsync, int ba
 
     if (!synchronous || !_response || _reference->getInvocationTimeout() > 0ms)
     {
-        auto stream = make_shared<InputStream>();
+        auto stream = make_shared<InputStream>(_reference->getInstance().get());
```

Comment:
I guess this is safe because the instance always out-lives this streams.

Reply:
It should, yes.